src/adapter/src/coord/sequencer/inner/peek.rs

```diff
@@ -1124,19 +1138,7 @@ impl Coordinator {
                     oracle_read_ts,
                     real_time_recency_ts,
                 )?;
-                // We only need read holds if the read depends on a timestamp.
-                let read_holds = match determination.timestamp_context.timestamp() {
-                    Some(_ts) => Some(read_holds),
-                    None => {
-                        // We don't need the read holds and shouldn't add them
-                        // to the txn.
-                        //
-                        // TODO: Handle this within determine_timestamp.
-                        drop(read_holds);
-                        None
-                    }
-                };
```

Comment:
I removed this part to make queries like this work:
```sql
CREATE VIEW v AS SELECT 1;
CREATE DEFAULT INDEX ON v;
SELECT * FROM v;
```

`implement_peek_plan` looks for read holds to pass to `create_dataflow` in the txn read holds and panics if it doesn't find them, so we need to put them in at some point. We don't have to add them here, but I don't see any reason not to. It removes a special case and probably doesn't have any negative impact?